FeedAPIChallenge/Feed API/RemoteFeedLoader.swift

```diff
@@ -19,6 +40,26 @@ public final class RemoteFeedLoader: FeedLoader {
 	}
 
 	public func load(completion: @escaping (FeedLoader.Result) -> Void) {
-		fatalError("Must be implemented")
+		client.get(from: self.url) { [weak self] result in
+			guard self != nil else { return }
+
+			guard let (data, httpResponse) = try? result.get() else {
+				return completion(.failure(Error.connectivity))
+			}
+
+			completion(RemoteFeedLoader.map(data, from: httpResponse))
+		}
+	}
+
+	private static func map(_ data: Data, from httpResponse: HTTPURLResponse) -> FeedLoader.Result {
+		guard httpResponse.statusCode == 200 else {
+			return .failure(Error.invalidData)
+		}
+
+		guard let response = FeedImageResponse.decode(from: data) else {
+			return .failure(Error.invalidData)
+		}
```

Comment:
Since both `guard` statements return with the same type of `failure`, consider joining them:
```suggestion
		guard httpResponse.statusCode == 200,
		      let response = FeedImageResponse.decode(from: data) else {
			return .failure(Error.invalidData)
		}
```